pkg/storage/streams/store.go

```diff
@@ -84,19 +94,31 @@ func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	var lastSegmentSize int64
 
 	awareLimitReader := EOFAwareReader(data)
+	encKey := sha256.Sum256([]byte(s.key))
+	var firstNonce [12]byte
 
 	for !awareLimitReader.isEOF() {
+		// TODO (moby) should I create a new encrypter for each segment?
+		encrypter, err := eestream.NewAESGCMEncrypter(&encKey, &firstNonce, s.encryptionBlockSize)
+		if err != nil {
+			return Meta{}, err
+		}
+
 		segmentPath := path.Prepend(fmt.Sprintf("s%d", totalSegments))
 		segmentData := io.LimitReader(awareLimitReader, s.segmentSize)
+		paddedData := eestream.PadReader(ioutil.NopCloser(segmentData), encrypter.InBlockSize())
+		transformedData := eestream.TransformReader(paddedData, encrypter, 0)
 
-		putMeta, err := s.segments.Put(ctx, segmentPath, segmentData,
+		putMeta, err := s.segments.Put(ctx, segmentPath, transformedData,
 			nil, expiration)
 		if err != nil {
 			return Meta{}, err
 		}
 		lastSegmentSize = putMeta.Size
 		totalSize = totalSize + putMeta.Size
 		totalSegments = totalSegments + 1
+
+		// TODO (moby) update nonce
```

Comment:
oh, if you do this you can ignore my comment about the thing with the nonces and security hazards